src/net_processing.cpp

```diff
@@ -4194,7 +4192,9 @@ bool PeerManagerImpl::SendMessages(CNode* pto)
     // If we get here, the outgoing message serialization version is set and can't change.
     const CNetMsgMaker msgMaker(pto->GetCommonVersion());
 
-    MaybeSendPing(*pto, *peer);
+    const auto current_time = GetTime<std::chrono::microseconds>();
+
+    MaybeSendPing(*pto, *peer, current_time);
```

Comment:
super nit: `current_time` here is `const`. Not sure if the intention for `now` is to be immutable in `MaybeSendPing()`, but I think  `now` would still be mutable even though the outer scoped variable is `const` because of the param signature in `MaybeSendPing(..., std::chrono::microseconds now)`

Reply:
In general, there's no benefit to making pass by value parameters const. Passing by reference to const indicates to the caller that the parameter will not be mutated. When passing by value, a copy of the parameter is made, so the original object cannot be mutated.

There's very good documentation about how to pass parameters in the c++ core guidelines: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f16-for-in-parameters-pass-cheaply-copied-types-by-value-and-others-by-reference-to-const.

Reply:
Thanks, apologies, I think I should have been more clear. Even though the `std::chrono` is passed by value, MaybeSendPing() treats it as mutable even though I don't think the method mutates the copy. So I was just thinking would it be useful/or maybe unecessary to make current_time in MaybeSendPing() also const.

Reply:
I understand. Yes, it is possible to mark the function parameters as `const` in the definition, which would prevent the copy from being mutated within the function body. You can even do this if the function declaration does not mark the parameter as `const`.

However, I haven't ever seen that style used in this project. It may be marginally useful, but it may also be misleading (eg see https://groups.google.com/u/4/g/comp.lang.c++.moderated/c/DEQTho8OCj8).